docs/tutorial.rst

```diff
@@ -1,4 +1,620 @@
+         let formData = new FormData(document.querySelector('form'));
+         let response = await fetch('/show', {
+           method: 'POST',
+           body: JSON.stringify(Object.fromEntries(formData))
+         });
+         let json = await response.json();
+         document.getElementById('src_dt').innerHTML = json['src_dt']['text'];
+         document.getElementById('src_dt').setAttribute('datetime', json['src_dt']['value']);
+         document.getElementById('src_location').innerHTML = json['src_location'];
+         document.getElementById('dst_dt').innerHTML = json['dst_dt']['text'];
+         document.getElementById('dst_dt').setAttribute('datetime', json['dst_dt']['value']);
+         document.getElementById('dst_location').innerHTML = json['dst_location'];
+         document.querySelector('.info').style.display = "block";
+       }
+     </script>
+   </head>
+   <body>
+     <h1>Time zone convertor</h1>
+     <form action="." method="post">
+       <input type="datetime-local" name="dt" value="{now}" required>
+
+       <div class="timezones">
+         <div class="timezone">
+           <label for="src">From:</label>
+           <select name="src" id="src">
+             {#zones}
+             {@eq key=location value="{default_src}"}
+             <option value="{zone}" selected>{location}</option>
+             {:else}
+             <option value="{zone}">{location}</option>
+             {/eq}
+             {/zones}
+           </select>
+         </div>
+
+         <div class="timezone">
+           <label for="dst">To:</label>
+           <select name="dst" id="dst">
+             {#zones}
+             {@eq key=location value="{default_dst}"}
+             <option value="{zone}" selected>{location}</option>
+             {:else}
+             <option value="{zone}">{location}</option>
+             {/eq}
+             {/zones}
+           </select>
+         </div>
+       </div>
+     </form>
+     <p><button onclick="showResult()">Show</button></p>
+
+     <p class="info">
+       When it's <time id="src_dt" datetime="2020-01-01T18:00">Jan 1 2020</time>
+       in <span id="src_location">UTC</span>,<br>
+       it's <time id="dst_dt" datetime="2020-01-01T18:00">Jan 1 2020</time>
+       in <span id="dst_location">UTC</span>.
+     </p>
+   </body>
+   </html>
+
+
+The changes are:
+
+- The template for showing the result has been merged.
+  It contains dummy information.
+
+- The JavaScript code for updating the page is added.
+  It gets called when the button is clicked.
+
+- The button element is moved out of the form element.
+  Also, the form action now points to the current URL.
+
+One last thing to do is to hide the result markup
+before the user clicks the "Show" button.
+This can be easily achieved in CSS:
+
+.. code-block:: css
+
+   p.info {
+     display: none;
+   }
+
+
+This concludes the introductory tutorial.
+Check out the other documents and example applications
+for advanced usage of Clastic features.
```

Comment:
I think here would be a good place to link to the complete application in the repo, and maybe include a screenshot (even if it's the same one from the top).

Reply:
The link is done. I'm not sure about repeating the screenshot, though.

Reply:
I've found redundancy isn't too bad, in the same way you've got the full code and then the segments. Totally up to you and we can add it anytime, after we launch and see how it looks even :)